cloudinit/config/schemas/schema-cloud-config-v1.json

```diff
@@ -866,85 +866,172 @@
               "default": "ansible"
             },
             "pull": {
-              "required": [
-                "url",
-                "playbook_name"
-              ],
-              "type": "object",
-              "additionalProperties": false,
-              "properties": {
-                "accept_host_key": {
-                  "type": "boolean",
-                  "default": false
-                },
-                "clean": {
-                  "type": "boolean",
-                  "default": false
-                },
-                "full": {
-                  "type": "boolean",
-                  "default": false
-                },
-                "diff": {
-                  "type": "boolean",
-                  "default": false
-                },
-                "ssh_common_args": {
-                  "type": "string"
-                },
-                "scp_extra_args": {
-                  "type": "string"
-                },
-                "sftp_extra_args": {
-                  "type": "string"
-                },
-                "private_key": {
-                  "type": "string"
-                },
-                "checkout": {
-                  "type": "string"
-                },
-                "module_path": {
-                  "type": "string"
-                },
-                "timeout": {
-                  "type": "string"
-                },
-                "url": {
-                  "type": "string"
-                },
-                "connection": {
-                  "type": "string"
-                },
-                "vault_id": {
-                  "type": "string"
-                },
-                "vault_password_file": {
-                  "type": "string"
-                },
-                "verify_commit": {
-                  "type": "boolean",
-                  "default": false
-                },
-                "inventory": {
-                  "type": "string"
-                },
-                "module_name": {
-                  "type": "string"
-                },
-                "sleep": {
-                  "type": "string"
-                },
-                "tags": {
-                  "type": "string"
-                },
-                "skip_tags": {
-                  "type": "string"
+              "oneOf": [
+                {
+                  "required": [
```

Comment:
Let's avoid duplicating this entire definition and using it twice in the array and object items of this oneOf. The way we do that is by defining a new $defs/ansible.pull object which can be used in both the array case and the object case.